modules/payment/commerce_payment.module

```diff
@@ -75,6 +78,24 @@ function commerce_payment_theme() {
 }
 
 /**
+ * Implements hook_theme_suggestions_HOOK().
+ */
+function commerce_payment_theme_suggestions_commerce_payment_gateway_form(array $variables) {
+  $original = $variables['theme_hook_original'];
+  $operation = strtr($variables['form']['#operation'], '-', '_');
+  /** @var \Drupal\commerce_payment\Entity\PaymentMethodInterface $default_payment_method */
+  $default_payment_method = $variables['form']['#default_value'];
+  $plugin_id = $default_payment_method->getPaymentGatewayId();
+
+  $suggestions = [];
+  $suggestions[] = $original . '__' . $operation;
+  $suggestions[] = $original . '__' . $plugin_id;
+  $suggestions[] = $original . '__' . $plugin_id . '__' . $operation;
```

Comment:
Do we want to add the instance id here, too?  I didn't see it in the form structure.